Marsiple/Common/Network/DataFetcher.swift

```diff
@@ -0,0 +1,52 @@
+//
+//  DataFetcher.swift
+//  Marsiple
+//
+//  Created by Matej Korman on 23/07/2018.
+//  Copyright © 2018 Matej Korman. All rights reserved.
+//
+
+import Foundation
+
+class DataFetcher {
```

Comment:
I like what you did here 👍 

Do you see a possibility for making the `DataFetcher` fully generic?

Reply:
You mean like this `class DataFetcher<DataType: Codable>`?

Reply:
I do, yeah.

Reply:
After thinking about it for a while yes, but I'm not quite sure how to do this. I think i need just a bit more time to figure it out.

Reply:
This is not a trivial solution, but is a good example of what it takes to trully make generics work. We can leave this as an improvement in the future, you don't have to be blocked by this.